blocks/initialization.py

```diff
@@ -223,3 +223,23 @@ def generate(self, rng, shape):
                                                  replace=False)
             weights[i, random_indices] = values[i]
         return weights
+
+
+class Xavier(NdarrayInitialization):
+    """Initialize with Gaussian distribution with Xavier parameters.
+
+    Use the following gaussian parameters: mean=0 and var=scale/Nin
```

Comment:
Apparently you forgot a square root here. Also we would need a reference to the paper here.